src/server/bitops_family.cc

```diff
@@ -57,14 +65,88 @@ constexpr int32_t GetByteIndex(uint32_t offset) noexcept {
   return offset / OFFSET_FACTOR;
 }
 
-uint8_t GetByteValue(const std::string& str, uint32_t offset) {
+uint8_t GetByteValue(std::string_view str, uint32_t offset) {
   return static_cast<uint8_t>(str[GetByteIndex(offset)]);
 }
 
 constexpr bool CheckBitStatus(uint8_t byte, uint32_t offset) {
   return byte & (0x1 << offset);
 }
 
+constexpr std::uint8_t CountBitsRange(std::uint8_t byte, std::uint8_t from, uint8_t to) {
+  int count = 0;
+  for (int i = from; i < to; i++) {
+    count += CheckBitStatus(byte, GetNormalizedBitIndex(i));
+  }
+  return count;
+}
+
+constexpr std::uint8_t BitOnCount(std::uint8_t byte) {
+  return NIBBLE_LOOKUP[byte & 0x0F] + NIBBLE_LOOKUP[byte >> 4];
+}
+
+std::uint32_t CountSetAllBytes(std::string_view at, int32_t start, int32_t end) {
+  if (start >= end || start < 0 || end < 0) {
+    return 0;
+  }
+  end = std::min(end, static_cast<int32_t>(at.size()));  // don't overflow
+  std::uint32_t count = std::accumulate(
+      std::next(at.begin(), start), std::next(at.begin(), end), 0,
+      [](auto counter, auto ch) { return counter + BitOnCount(static_cast<std::uint8_t>(ch)); });
```

Comment:
again, cast is redundant. Please use size_t for counting. I would choose a single loop that will take the same number lines like this bombastic accumulate and will be 3 times more readable, IMHO. 

And btw, it's possible to improve the performance of this function by factor of 8 if you would iterate by chunks of 8 bytes and pass `uint64_t` into `absl::popcount`.

Reply:
I though about this, Wasn't sure that it would really make it faster in the real world, so dropped it.
I remove the cast along with removing "BitOnCount" function